### ISSUE TITLE
axis parameters for backend functions

### ISSUE THREAD

**Issue by jjallaire** [rstudio/keras#202]
I'm implementing R wrappers for the `K$backend` functions, for example:

```r
#' Concatenates a list of tensors alongside the specified axis.
#' 
#' @param tensors list of tensors to concatenate.
#' @param axis concatenation axis.
#' 
#' @return A tensor.
#' 
#' @export
k_concatenate <- function(tensors, axis = -1L) {
  keras$backend$concatenate(
    tensors = tensors,
    axis = as_axis(axis)
  )
}
```

This will provide improved help as well as automatic handling of integer and other required type conversions.

A question I have related to this is what should the baseline for `axis` parameters be? We have moved to 1-based indexing for extraction `[` in TensorFlow so it seems like we should also use 1-based indexes for referring to axes (as is done in R functions like `sweep()` and `apply()`. Do people agree with this decision?

I'm also wondering what value I should use to refer to "all axes". As you can see in the snippet above Python uses `-1L` for this. I'm inclined to also use `-1L` for R as I think that `0` would be confusing for anyone aware that 0-based indexing is possible. Do people agree with this?

cc @kevinykuo @kevinushey @terrytangyuan @goldingn @dfalbel

**Comment by terrytangyuan** [comment]
Yes, I agree with using -1L. 0 looks odd to me. Also agree with 1-based indexing to align with apply().

**Comment by goldingn** [comment]
Yep, that sounds sensible,
